velox/common/base/BitUtil.h

```diff
@@ -977,6 +979,25 @@ inline __int128_t builtin_bswap128(__int128_t value) {
 #endif
 }
 
+/// Store `bits' into the memory region pointed by `byte', at `index' (bit
+/// index).  If `kSize' is 8, we store the whole byte directly; otherwise it
+/// must be 4 and we store either the whole byte or the upper 4 bits only,
+/// depending on the `index'.
+template <int kSize>
+void storeBitsToByte(uint8_t bits, uint8_t* bytes, unsigned index) {
```

Comment:
Let's add a unit test for this function to help protect the logic and illustrate usage.

Reply:
Added